packages/connection/src/Manager.php

```diff
@@ -43,6 +193,313 @@ public function initialize( $methods ) {
+	 * from /xmlrpc.php so that we're replicating it as closely as possible.
+	 */
+	public function alternate_xmlrpc() {
+		// phpcs:disable PHPCompatibility.Variables.RemovedPredefinedGlobalVariables.http_raw_post_dataDeprecatedRemoved
+		// phpcs:disable WordPress.WP.GlobalVariablesOverride.Prohibited
+		global $HTTP_RAW_POST_DATA;
+
+		// Some browser-embedded clients send cookies. We don't want them.
+		$_COOKIE = array();
+
+		// A bug in PHP < 5.2.2 makes $HTTP_RAW_POST_DATA not set by default,
+		// but we can do it ourself.
+		if ( ! isset( $HTTP_RAW_POST_DATA ) ) {
+			$HTTP_RAW_POST_DATA = file_get_contents( 'php://input' );
+		}
+
+		// A fix for mozBlog and other cases where '<?xml' isn't on the very first line.
+		if ( isset( $HTTP_RAW_POST_DATA ) ) {
+			$HTTP_RAW_POST_DATA = trim( $HTTP_RAW_POST_DATA );
+		}
+
+		// phpcs:enable
+
+		include_once ABSPATH . 'wp-admin/includes/admin.php';
+		include_once ABSPATH . WPINC . '/class-IXR.php';
+		include_once ABSPATH . WPINC . '/class-wp-xmlrpc-server.php';
+
+		/**
+		 * Filters the class used for handling XML-RPC requests.
+		 *
+		 * @since 3.1.0
+		 *
+		 * @param string $class The name of the XML-RPC server class.
+		 */
+		$wp_xmlrpc_server_class = apply_filters( 'wp_xmlrpc_server_class', 'wp_xmlrpc_server' );
+		$wp_xmlrpc_server       = new $wp_xmlrpc_server_class();
+
+		// Fire off the request.
+		nocache_headers();
+		$wp_xmlrpc_server->serve_request();
+
+		exit;
+	}
+
+	/**
+	 * Removes all XML-RPC methods that are not `jetpack.*`.
+	 * Only used in our alternate XML-RPC endpoint, where we want to
+	 * ensure that Core and other plugins' methods are not exposed.
+	 *
+	 * @param array $methods a list of registered WordPress XMLRPC methods.
+	 * @return array filtered $methods
+	 */
+	public function remove_non_jetpack_xmlrpc_methods( $methods ) {
+		$jetpack_methods = array();
+
+		foreach ( $methods as $method => $callback ) {
+			if ( 0 === strpos( $method, 'jetpack.' ) ) {
+				$jetpack_methods[ $method ] = $callback;
+			}
+		}
+
+		return $jetpack_methods;
+	}
+
+	/**
+	 * Removes all other authentication methods not to allow other
+	 * methods to validate unauthenticated requests.
+	 */
+	public function require_jetpack_authentication() {
+		// Don't let anyone authenticate.
+		$_COOKIE = array();
+		remove_all_filters( 'authenticate' );
+		remove_all_actions( 'wp_login_failed' );
+
+		if ( $this->is_active() ) {
+			// Allow Jetpack authentication.
+			add_filter( 'authenticate', array( $this, 'authenticate_jetpack' ), 10, 3 );
+		}
+	}
+
+	/**
+	 * Authenticates XML-RPC and other requests from the Jetpack Server
+	 *
+	 * @param WP_User|Mixed $user user object if authenticated.
+	 * @param String        $username username.
+	 * @param String        $password password string.
+	 * @return WP_User|Mixed authenticated user or error.
+	 */
+	public function authenticate_jetpack( $user, $username, $password ) {
+		if ( is_a( $user, '\WP_User' ) ) {
```

Comment:
We need a double backslash here because of how they are parsed within single quotes.

```suggestion
		if ( is_a( $user, '\\WP_User' ) ) {
```